csp_billing_adapter/bill_utils.py

```diff
@@ -214,32 +214,48 @@ def get_volume_dimensions(
 
 def get_billing_dimensions(
     config: Config,
-    billable_usage: dict
+    billable_usage: dict,
+    status: dict = None
 ) -> dict:
     """
     Construct a hash mapping each metric's appropriate billable
     dimension name to it's usage value in the 'billable_usage'
     hash, using the metric's dimension configuration settings
-    from the provided 'config'.
+    from the provided 'config'. If a status dictionary is provided
+    the previous metering attempt partially failed. Any metrics
+    that billed successfully will be skipped.
 
     :param config:
         The configuration specifying the metrics that need to be
         processed in the usage records list.
     :param billable_usage:
         A hash mapping the usage metrics specific in the 'config' to
         their calculated usage values.
+    :param status:
+        A hash mapping the previous metering attempt status and
+        record ids. If the previous metering attempt was successful
+        the hash will be empty.
     :return:
         A hash mapping each metric tier's name to it's associated
         usage value.
     """
     billed_dimensions = {}
 
+    if not status:
```

Comment:
Minor suggestion; in my experience it is best to put any argument defaulting as the initial block of code inside a function/method, optionally with a comment to that effect, so that over time the significance doesn't get lost...

Reply:
Also if being pedantic, the check here should be `if status is None` to match the specified default value... With this check (`if not status`) any "false" value passed in will be converted to an empty dict (`{}`) which could potentially hide other errors, e.g. some code path passes in an empty list data structure that evaluates to false.